File: backend/tests/test_analysis.py
"""Test file for code analysis"""

def insecure_function(user_input):
    """This function has intentional security issues"""
    # Security issue: using eval
    result = eval(user_input)
    
    # Unused import
    import json
    
    # Undefined variable
    print(undefined_var)
    
    return result

# Missing docstring
def another_function():
    pass

# SQL Injection vulnerability
def query_db(user_input):
    query = "SELECT * FROM users WHERE id = " + user_input
    return query

File: backend/tests/test_analysis.js
// Test file for JavaScript analysis

// Security issue: eval usage
function insecureFunction(userInput) {
    return eval(userInput);
}

// Undefined variable
function buggyFunction() {
    console.log(undefinedVar);
}

// SQL Injection vulnerability
function queryDb(userInput) {
    const query = "SELECT * FROM users WHERE id = " + userInput;
    return query;
}

// XSS vulnerability
function displayUserInput(input) {
    document.innerHTML = input;
}